File: abc/abc131/abc131_d.fs
// https://atcoder.jp/contests/abc131/submissions/12885032

let N = stdin.ReadLine() |> int
let AB = Array.zeroCreate<int []> N

for i in [ 0..N - 1 ] do
    AB.[i] <- stdin.ReadLine().Split() |> Array.map int

let sAB = AB |> Array.sortBy (fun x -> x.[1])

let rec solver t n =
    if n = N then "Yes"
    else
        let time = t + sAB.[n].[0]
        if time > sAB.[n].[1] then "No"
        else solver time (n + 1)

solver 0 0 |> stdout.WriteLine

File: abc/abc131/abc131_d.cs
// https://atcoder.jp/contests/abc131/submissions/12883543

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var N = int.Parse(Console.ReadLine());

        var AB = new int[N][];
        for (int i = 0; i < N; i++)
        {
            AB[i] = Console.ReadLine().Split().Select(int.Parse).ToArray();
        }
        Array.Sort(AB, (x, y) => x[1] - y[1]);

        var flag = true;
        var time = 0;
        for (int i = 0; i < N; i++)
        {
            if ((time += AB[i][0]) > AB[i][1])
            {
                flag = false;
                break;
            }
        }
        Console.WriteLine(flag ? "Yes" : "No");
    }
}
